internal/core/data/event.go

```diff
@@ -238,6 +238,7 @@ func updateEventPushDate(id string, ctx context.Context) error {
 func putEventOnQueue(evt models.Event, ctx context.Context) {
 	LoggingClient.Info("Putting event on message queue")
 
+	evt.CorrelationId = correlation.FromContext(ctx)
```

Comment:
I appreciate this emerging pattern (assuring CorrelationId to potentially include in error log).
To better ensure consistency across data models (fail fast earlier, etc)... should this be enforced when validating schema [to occur during json.Marshal] or would that open a can of worms?

Reply:
I think it depends. Like we're discussing below, in this use case the payload is an event but it could be something else. If over time we find that we're only using this mechanism for one thing, then we can harden it as our discussions warrant.

One thing I have in the back of my mind is the discussion in Seoul about notifying northside partners about changes to the devices in the environment. The Messaging module could be a way to solve that, but I suggest we wait and see.

Reply:
I largely agree. If we wait for CorrelationId to become a required property across a large percentage of contract data models, my concern was that defers a potentially significant # of breaking changes.